massa-client/src/cmds.rs

```diff
@@ -273,503 +276,491 @@ impl Command {
                 }
             }
+            Ok(Box::new(()))
+        }
 
-            Command::wallet_add_private_keys => {
-                let addresses = parse_vec::<PrivateKey>(parameters)?
-                    .into_iter()
-                    .map(|key| Ok((wallet.add_private_key(key)?, key)))
-                    .collect::<Result<HashMap<Address, PrivateKey>>>()?;
-                if json {
-                    return Ok(Box::new(addresses.into_keys().collect::<Vec<Address>>()));
-                } else {
-                    for (address, key) in addresses.iter() {
-                        println!("Derived and added address {} to the wallet.", address);
-                        println!("Type `node_add_staking_private_keys {}` to start staking with this private_key.\n", key);
+        Command::wallet_remove_addresses => {
+            let mut res = "".to_string();
+            for key in parse_vec::<Address>(parameters)?.into_iter() {
+                match wallet.remove_address(key) {
+                    Ok(_) => {
+                        res.push_str(&format!("Removed address {} from the wallet\n", key));
+                    }
+                    Err(WalletError::MissingKeyError(_)) => {
+                        res.push_str(&format!("Address {} wasn't in the wallet\n", key));
+                    }
+                    Err(_) => {
+                        res.push_str(&format!(
+                            "Failed to remove address {} from the wallet\n",
+                            key
+                        ));
                     }
                 }
-                Ok(Box::new(()))
             }
+            if !json {
+                println!("{}", res);
+            }
+            Ok(Box::new(()))
+        }
 
-            Command::wallet_remove_addresses => {
-                let mut res = "".to_string();
-                for key in parse_vec::<Address>(parameters)?.into_iter() {
-                    match wallet.remove_address(key) {
-                        Ok(_) => {
-                            res.push_str(&format!("Removed address {} from the wallet\n", key));
-                        }
-                        Err(WalletError::MissingKeyError(_)) => {
-                            res.push_str(&format!("Address {} wasn't in the wallet\n", key));
-                        }
-                        Err(_) => {
-                            res.push_str(&format!(
-                                "Failed to remove address {} from the wallet\n",
-                                key
-                            ));
-                        }
-                    }
-                }
-                if !json {
-                    println!("{}", res);
-                }
-                Ok(Box::new(()))
+        Command::buy_rolls => {
+            if parameters.len() != 3 {
+                bail!("wrong number of parameters");
             }
+            let addr = parameters[0].parse::<Address>()?;
```

Comment:
Since we are making a client that will be use by people that don't know code at all we should catch error message to re-send more user friendly ones ?